invenio_files_rest/cli.py

```diff
@@ -0,0 +1,104 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Click command-line interface for file management."""
+
+from __future__ import absolute_import, print_function
+
+import os
+
+import click
+from flask import current_app
+from flask_cli import with_appcontext
+from invenio_db import db
+
+#
+# File management commands
+#
+
+
+@click.group()
+def files():
+    """File management commands."""
+
+
+@files.command()
+@click.argument('source', type=click.Path(exists=True, resolve_path=True))
+@click.option('--checksum/--no-checksum', default=False)
+@with_appcontext
+def bucket_from(source, checksum):
+    """Create new bucket from all files in directory."""
+    from .helpers import compute_md5_checksum
+    from .models import Bucket, FileInstance, ObjectVersion
+
+    bucket = Bucket.create()
+
+    def create_file(key, path):
+        """Create new ``ObjectVersion`` from path or existing ``FileInstance``.
+
+        It checks MD5 checksum and size of existing ``FileInstance``s.
+        """
+        if checksum:
+            file_checksum = compute_md5_checksum(open(path, 'rb'),
+                                                 chunk_size=80960)
+            file_instance = FileInstance.query.filter_by(
+                checksum=file_checksum, size=os.path.getsize(path)
+            ).first()
+            if file_instance:
+                click.echo('Found same file instance "{0}" for "{1}".'.format(
+                    str(file_instance), path
+                ))
+                return ObjectVersion.create(
+                    bucket, key, file_id=file_instance.id
+                )
+        click.echo('Creating file "{0}" from "{1}".'.format(key, path))
+        return ObjectVersion.create(bucket, key, stream=open(path, 'rb'))
+
+    if os.path.isfile(source):
+         create_file(os.path.basename(source), source)
+    else:
+        for root, dirs, files in os.walk(source, topdown=False):
+            for name in files:
+                filename = os.path.join(root, name)
+                assert filename.startswith(source)
+                parts = [p for p in filename[len(source):].split(os.sep) if p]
+                create_file('/'.join(parts), os.path.join(root, name))
+
+    bucket.locked = True
+    db.session.commit()
+    click.secho(str(bucket), fg='green')
+
+
+@files.command()
+@click.argument('name')
+@click.argument('uri')
+@click.option('--default', is_flag=True, default=False)
+@with_appcontext
+def location(name, uri, default):
+    """Create new location."""
+    from .models import Location
+    location = Location(name=name, uri=uri, default=default)
```

Comment:
The previous default location should be unset. Perhaps implemented via a ``Location.create()``?